### DISCUSSION TITLE
How do I add a custom download button?

### DISCUSSION THREAD

**Discussion by gajraj-gan** [vidstack/player#963] [General]
Same as title, as far as I have read the docs we can access the data of video using the hooks but how do i add a custom button for dowloading?

**Comment by mihar-22** [comment]
Create a download link like so:

```html
<a href="path_to_video_file" download="video_file_name" aria-label="Download video">
  <!-- icon here. -->
</a>
```

**Reply by gajraj-gan** [reply]
How do i add it in the player? I am using React/NextJS.

**Reply by mihar-22** [reply]
You have to build a custom UI, see our examples: 

https://github.com/vidstack/examples

**Reply by mihar-22** [reply]
With our [new layout slots](https://www.vidstack.io/docs/player/components/layouts/default-layout?styling=default-theme#slots) for React, you can now easily insert/replace content inside our Default Layout. That should make this type of thing much easier for anyone in the future.

**Reply by gajraj-gan** [reply]
Thanks, it is perfect :)

**Comment by gajraj-gan** [comment]
I did this.

```js
<DefaultVideoLayout
          icons={defaultLayoutIcons}
          slots={{
            beforeSettingsMenu: (
              <button type="button" onClick={() => window.open(src, '_blank')}>
                <DownloadIcon color="white" size={32} />
              </button>
            )
          }}
        />
```
